presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1603,8 +1606,12 @@ private void executeAddPartitionOperations()
 
         private void executeUpdateStatisticsOperations()
         {
+            ImmutableList.Builder<CompletableFuture<?>> executeUpdateFutures = ImmutableList.builder();
             for (UpdateStatisticsOperation operation : updateStatisticsOperations) {
-                operation.run(delegate);
+                executeUpdateFutures.add(CompletableFuture.runAsync(() -> operation.run(delegate), updateExecutor));
+            }
+            for (CompletableFuture<?> executeUpdateFuture : executeUpdateFutures.build()) {
+                MoreFutures.getFutureValue(executeUpdateFuture, PrestoException.class);
```

Comment:
Previously failed operation would cause fast-failure of stats update. Now update would continue (in threads) even though one of the futures failed. I think we should wait for all futures to complete. Perhaps we should have a utility method that encapsulates mechanism from `executeIrreversibleMetastoreOperations`

WDYT: @electrum @findepi ?

Reply:
Good catch. We could use eg CompletionService to detect failure early, but yes, we still should somehow notify others need to be cancelled?

For stats happening during CREATE TABLE, it's fine to just cancel them (& await), as table creation rollback will clean the state.
Not sure what we should do for ANALYZE or ALTER.

Reply:
@findepi  are you suggesting that we cancel the other threads if one of the future fails or should we just wait for all futures to complete? If we just want to wait then we could do something like `executeIrreversibleMetastoreOperations` as @sopel39  suggested.

Reply:
If we could cancel remaining futures that would be best, but I'm not sure it's simple to implement. IMO something like `executeIrreversibleMetastoreOperations` is good enough.

Reply:
added changes similar to `executeIrreversibleMetastoreOperations`